### PR TITLE
Download photo

### PR DESCRIPTION
**Pull request by khoidt**

### CODE REVIEW

**Comment by khoidt** [comment]
[Add Dowload Photo to Download list in Fragmentarium](https://trello.com/c/AtE3XCpu/617-add-dowload-photo-to-download-list-in-fragmentarium)

File: src/fragmentarium/ui/fragment/Download.tsx [hunk 1]
```diff
@@ -2,23 +2,30 @@ import React, { useEffect, useState } from 'react'
 import { Fragment } from 'fragmentarium/domain/fragment'
 import * as TeiExport from 'fragmentarium/ui/fragment/TeiExport'
 import WordService from 'dictionary/application/WordService'
+import FragmentService from 'fragmentarium/application/FragmentService'
 import WordDownloadButton from 'fragmentarium/ui/fragment/WordDownloadButton'
 import PdfDownloadButton from 'fragmentarium/ui/fragment/PdfDownloadButton'
 import Download from 'common/Download'
+import { Promise } from 'bluebird'
+
+//import withData from 'http/withData'
```
Comment (fsimonjetz) [inline src/fragmentarium/ui/fragment/Download.tsx:11]: Leftover comment

File: src/fragmentarium/ui/fragment/Download.tsx [hunk 2]
```diff
@@ -59,12 +67,23 @@ export default function DownloadFragment({
     )
     setAtf(atfUrl)
 
+    let photoUrl = ''
+    if (fragment.hasPhoto && !photo) {
+      Promise.resolve(fragmentService.findPhoto(fragment)).then((photo) => {
+        photoUrl = URL.createObjectURL(photo)
+        setPhoto(photoUrl)
```
Comment (fsimonjetz) [inline src/fragmentarium/ui/fragment/Download.tsx:74]: I think if the condition is true, `setPhoto(photoUrl)` is called twice – maybe put the second call inside an `else` block?

File: src/fragmentarium/ui/fragment/Download.test.tsx [hunk 3]
```diff
@@ -40,19 +56,24 @@ describe.each([
...
   ['Download as TEI XML File', 'xml', teiUrl],
-])('%s download link', (name, type, url) => {
+  ['Download Photo', 'jpg', photoUrl],
+])('%s download link', (name: string, type: string, url: string) => {
   test('href', () => {
-    expect(screen.getByRole('link', { name: `${name}` })).toHaveAttribute(
-      'href',
-      url
-    )
+    if (type !== 'jpg' || fragment.hasPhoto) {
```
Comment (fsimonjetz) [inline src/fragmentarium/ui/fragment/Download.test.tsx:62]: I'm not sure if I follow the logic here, can you clarify @khoidt ?

**Review by fsimonjetz — APPROVED** [review]

**Pull request merged**